torch/_inductor/fx_passes/quantization.py

```diff
@@ -532,76 +543,100 @@ def __init__(
+                dtype=torch.bfloat16
+                if int8_mixed_bf16_with_inplace_add
+                else torch.float32,
+            ),
+            BinaryUnaryAttr(
+                "add", 1.0, "relu", [], ""
+            ): generate_pattern_with_output_quant(
+                generate_pattern_with_unary(
+                    generate_pattern_with_binary(
+                        aten.add.Tensor,
+                        dequantize_qconv_pt2e_pattern,
+                        dequantize_accum_pattern,
+                        int8_mixed_bf16_with_inplace_add,
+                    ),
+                    aten.relu.default,
+                ),
+                dtype=torch.bfloat16
+                if int8_mixed_bf16_with_inplace_add
+                else torch.float32,
+            ),
+        }
+
+        for binary_unary_attr, patterns in binary_replace_patterns.items():
+            _register_quantized_conv_binary_lowering(
+                patterns,
+                0,  # pass_number
+                torch.ops.onednn.qconv2d_pointwise.binary,  # computation_op
+                None,  # output_dtype
+                binary_unary_attr,  # binary_unary_attr
             )
-        ),
-    }
 
-    for binary_unary_attr, patterns in binary_replace_patterns.items():
-        _register_quantized_conv_binary_lowering(
+        # Priority 2 to match: QConv2d Binary-Unary pattern with fp32/bfloat16 output
+        binary_replace_float_out_patterns = {
+            BinaryUnaryAttr("add", 1.0, "relu", [], ""): generate_pattern_with_unary(
+                generate_pattern_with_binary(
+                    aten.add.Tensor,
+                    dequantize_qconv_pt2e_pattern,
+                    KeywordArg("accum_after_dequant"),
+                    int8_mixed_bf16_with_inplace_add,
+                ),
+                aten.relu.default,
+            ),
+        }
+
+        for (
+            binary_unary_attr,
             patterns,
-            0,  # pass_number
-            torch.ops.onednn.qconv2d_pointwise.binary,  # computation_op
-            None,  # output_dtype
-            binary_unary_attr,  # binary_unary_attr
-        )
+        ) in binary_replace_float_out_patterns.items():
+            _register_quantized_conv_binary_lowering(
+                patterns,
+                0 if int8_mixed_bf16_with_inplace_add else 1,  # pass_number
+                torch.ops.onednn.qconv2d_pointwise.binary,  # computation_op
+                # Note that for int8-mixed-bf16 and non-inplace add, because we have
+                # q-dq inserted at extra input of add, so the non-inplace add has bf16 and fp32 inputs,
+                # the output dtype will be float32.
+                # For inplace add, there is a extra to_bf16 node at add output, so the fusion pattern has bfloat16 output.
+                torch.bfloat16 if int8_mixed_bf16_with_inplace_add else torch.float32,
+                binary_unary_attr,  # binary_unary_attr
+            )
```

Comment:
it might be better just to have two separate patterns instead of multiple if/else

Reply:
Thanks and changed.